src/Angular.js

```diff
@@ -125,6 +128,39 @@ var VALIDITY_STATE_PROPERTY = 'validity';
 
 var hasOwnProperty = Object.prototype.hasOwnProperty;
 
+var objectMaxDepthInErrorMessage = 5;
+
+/**
+ * @ngdoc function
+ * @name angular.configureErrorHandling
+ * @module ng
+ * @kind function
+ *
+ * @description Use this function to configure error handling.
+ *
+ * @param {Object=} config An object for defining configuration for error handling. The
+ *     following keys are supported:
+ *
+ * * `objectMaxDepth` - used to stringify objects in error messages until reaching the max depth.
+ * if it is not supplied the default max depth which is 5 will be used instead, or if it is invalid
+ * (less than `1` or not a `Number`) no max depth will be used.
+ *
+ */
+function configureErrorHandling(config) {
+  if (isObject(config)) {
+    objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth : NaN;
+  }
+}
```

Comment:
With the prospect of #15590 being solved via this method, I think we need to revise the API to make it easily support what we need. Basically, we need it to support the following:

1. Getting the current values.
2. Setting specific values (so users don't have to specify all of them every time).

##
I suggest something along these lines:

```js
/**
 * @ngdoc function
 * @name angular.errorHandlingConfig
 * @module ng
 * @kind function
 *
 * @description
 * Configure several aspects of error handling in AngularJS if used as a setter or return the
 * current configuration if used as a setter. The following options are supported:
 *
 * - **objectMaxDepth**: The maximum depth to which objects are traversed when stringified for error
 *                       messages.
 * - **someOtherOption**: Blah, blah, blah.
 * - ...
 *
 * Omitted or undefined options will leave the corresponding configuration values unchanged.
 *
 * @param {Object=} config - The configuration object. MAy only contain the options that need to be
 *     updated. Supported keys:
 *     - `objectMaxDepth` **{Number}** - The max depth for stringifying objects. Setting to a
 *       non-positive or non-numeric value, removes the max depth limit.
 *       Default: 5
 *     - `someOtherOption` **{SomeType}** - Blah, blah, blah.
 *       Default: someDefaultValue
 *     - ...
 */
function errorHandlingConfig(config) {
  if (config) {
    return {
      objectMaxDepth: objectMaxDepthInErrorMessage
    };
  } else {
    if (isDefined(config.objectMaxDepth)) {
      objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth : NaN;
    }
    if (isDefined(config.someOtherOption)) {
      /* someOtherOption = config.someOtherOption */
    }
    ...
  }
}
```

WDYT?